x-pack/plugins/data_enhanced/server/search/session/session_service.ts

```diff
@@ -325,8 +296,22 @@ export class SearchSessionService implements ISessionService {
     );
   };
 
+  public extend(deps: SearchSessionDependencies, sessionId: string, keepAlive: string) {
+    this.logger.debug(`extend | ${sessionId}`);
+
+    // Calculate the new `expires` value given the `keepAlive`
+    const expiresMoment = dateMath.parse(`now+${keepAlive}`);
```

Comment:
Shouldn't the expiration be added on top of the current expiration date?

So if something expires in a week, and I extend it by another week, I should have an expiration of two weeks total.

Reply:
Hmm, that would probably make sense. I think we should discuss what our ideal API for this is. What happens currently when you click "extend"? Does it provide you an option of how long to extend it? Do we arbitrarily choose an amount to extend it by?

Reply:
I think the API should expect the client side to do all the math and pass in the final expiration in.

I think that if we have a session that expires in 1 week, and we wish to extend it to two, the API should receive `2w` and just pass it in to ES.

This will make changes more simple IMO. What do you think?